spec/services/supplier_chooser_spec.rb

```diff
@@ -1,23 +1,82 @@
 RSpec.describe SupplierChooser do
-  subject(:service) { described_class.new(doorkeeper_application_owner, from_location) }
+  subject(:service) { described_class.new(date, location) }
 
-  let(:from_location) { instance_double('Location', suppliers: [supplier_one, supplier_two]) }
-  let(:supplier_one) { instance_double('Supplier') }
-  let(:supplier_two) { instance_double('Supplier') }
+  let(:supplier1) { create(:supplier) }
+  let(:supplier2) { create(:supplier) }
+  let(:location) { create(:location) }
+  let(:date) { Date.today }
 
-  context 'when doorkeeper_application_owner is nil' do
-    let(:doorkeeper_application_owner) { nil }
+  context 'without an effective from or to date' do
+    let!(:supplier_location) { create(:supplier_location, supplier: supplier1, location: location) }
 
-    it 'returns the first supplier from move.from_location.suppliers' do
-      expect(service.call).to eq(supplier_one)
+    it 'returns matching supplier' do
+      expect(service.call).to eq(supplier1)
```

Comment:
sorry it's not clear why it is returning only supplier1 to me

Reply:
I updated the description in the spec, clearer now?